### ISSUE TITLE
No module named 'azureml.automl.runtime._ml_engine.featurizer_suggestion'

### ISSUE THREAD

**Issue by KaiquanMah** [Azure/MachineLearningNotebooks#1297]
When I ran AutoMLConfig with parameter featurization='auto', AutoMLConfig ran successfully.
Based on the successfully defined AutoMLConfig, I submitted it to an automl experiment called 'automl_run'.

Yet in my subsequent steps when I ran 'automl_run.get_output()', I encountered the error below:
ModuleNotFoundError: No module named 'azureml.automl.runtime._ml_engine.featurizer_suggestion'

It seems that AZ ML is unable to find the 'azureml.automl.runtime._ml_engine.featurizer_suggestion' module to run an experiment with AutoMLConfig featurization='auto'. Has the AutoMLConfig featurization functionality been deprecated?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0bc2b21e-6b1a-cb94-2857-147177a29d7c
* Version Independent ID: d14620a6-a2f6-49f1-632e-73903d41de8c
* Content: [azureml.train.automl.automlconfig.AutoMLConfig class - Azure Machine Learning Python](https://docs.microsoft.com/en-us/python/api/azureml-train-automl-client/azureml.train.automl.automlconfig.automlconfig?view=azure-ml-py)
* Content Source: [AzureML-Docset/stable/docs-ref-autogen/azureml-train-automl-client/azureml.train.automl.automlconfig.AutoMLConfig.yml](https://github.com/MicrosoftDocs/MachineLearning-Python-pr/blob/live/AzureML-Docset/stable/docs-ref-autogen/azureml-train-automl-client/azureml.train.automl.automlconfig.AutoMLConfig.yml)
* Service: **machine-learning**
* Sub-service: **core**
* GitHub Login: @DebFro
* Microsoft Alias: **debfro**

**Comment by marcindulak** [comment]
It's probably due to a mismatch between the azureml sdk on the jupyter instance and on the compute instance.

https://github.com/Azure/MachineLearningNotebooks/issues/842#issuecomment-595908014 says
 
`the automl version installed locally is the one that gets used for remote training`

Is this still the case?

Would a feature where an environment is passed to the automl run explicitly be feasible?

**Comment by jeff-shepherd** [comment]
Yes, this is caused by a mismatch between the version of the SDK used to training the model and the version of the SDK used to execute get_output.  Specifically, I can reproduce this issue by training the model with SDK version 1.20.0 and then running get_output from SDK version 1.19.0.

The issue can be resolved by upgrading the local SDK (the one used to execute get_output).  You can do this by creating a new Compute Instance or by running the automl_setup as described at [https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/automated-machine-learning/README.md](https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/automated-machine-learning/README.md)

The statement "the automl version installed locally is the one that gets used for remote training" no longer applies.  Instead, automl uses a curated environment by default for remote training.  This is based on the latest SDK version.  We might expose options to select the curated environment in the future.

You can also look at the sample notebook [https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/automated-machine-learning/experimental/regression-model-proxy/auto-ml-regression-model-proxy.ipynb](https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/automated-machine-learning/experimental/regression-model-proxy/auto-ml-regression-model-proxy.ipynb).
This uses remote inference.  It doesn't need to call get_output, so it is not as sensitive to version mismatches.

**Comment by CESARDELATORRE** [comment]
Closing the issue since it was unblocked by @jeff-shepherd.
Re-open if needed.

**Comment by jadhosn** [comment]
@CESARDELATORRE can you please re-open? I'm facing the same issue. Trained an AutoML model using the UI, and once I download it (also using UI), I'm unable to load the pickle because of `No module named 'azureml.automl.runtime'`

**Comment by lostmygithubaccount** [comment]
@jadhosn it would probably be best to open a new issue and link previous issue(s), apologies for the inconvenience

**Comment by jadhosn** [comment]
Actually, it was from my end. Once I recreated the environment (separate from my local env), I was able to unpickle it.
